lib/router.js

```diff
@@ -97,9 +98,17 @@ Router.prototype._buildPath = function route(node, path, value) {
 /**
  * Load and initialize a module
  */
-Router.prototype._loadModule = function(modDef, config) {
+Router.prototype._loadModule = function(modDef, globals) {
 
     var self = this;
+
+    // First, check if we have a copy of this module in the cache, so that we
+    // can share it.
+    var cachedModule = self._modules.get(modDef);
+    if (cachedModule && cachedModule._parentGlobals === globals) {
```

Comment:
@gwicke I've tried to put a `console.log` before the resolve call, and this sharing doesn't work. There's a whole bunch of problems with it:
- We don't specify the module type explicitly, but resolve it from the file extension and then modify the `modDef`. So the get call always uses
```javascript
{ path: 'some_path.js' }
```
as a key, while the set call always uses
```javascript
{ path: 'some_path.js', type: 'file' }
```
so the match is never found.
- The map keys equality are determined by the semantics of a `===` operator, so only references to the exact same object instance would be equal. In our case, the `modDef` objects a different instances with the same keys and values, so even if the object are deepEqual, the match wouldn't be found. We need to `JSON.stringify` the key.
- Yet another problem is with globals equality. `===` doesn't work as a deep equality on objects, so the second part of the condition would be almost always false.